pepper-apis/src/test/java/org/broadinstitute/ddp/route/PostPasswordResetRouteTest.java

```diff
@@ -56,27 +56,23 @@ public static void setup() {
 
     @Test
     public void test_WhenRouteIsCalledWithValidClientId_ItRespondsWithCorrectHttpRedirect() {
-        Map<String, String> queryParams = new HashMap<>();
-        queryParams.put(QueryParam.AUTH0_CLIENT_ID, auth0ClientId);
-        queryParams.put(QueryParam.AUTH0_DOMAIN, auth0Domain);
-        queryParams.put(QueryParam.EMAIL, testEmail);
-        queryParams.put(QueryParam.SUCCESS, "true");
-        HttpUrl fullUrl = buildEncodedUrl(url, queryParams);
         given(TestUtil.RestAssured.nonFollowingRequestSpec())
-                .when().get(fullUrl.toString()).then().assertThat()
```

Comment:
We were using `HttpUrl`, which was from a transitive dependency of `okhttp v3.9.1` that was pulled in by the `jslack` library. This version of `HttpUrl` did not do proper encoding of query parameters. After removing the `jslack` library, the version we're using became `okhttp v3.12.0` which came from the `google-maps-services` library. This new version of `HttpUrl` have updated rules for encoding query params.

What this all means is that:
* Previously, this test was NOT encoding the query params. So we pass in `fullUrl` which is not encoded. The `RestAssured` library will do a round of encoding itself, so nothing was wrong here. The `PostPasswordResetRoute` received properly encoded query params, SparkJava does the right thing and decodes it, and we're able to find the client and such. All good.
* However, now, `fullUrl` is getting encoded, and because `RestAssured` also does encoding, we have issue of **double encoding**. The end result is that in `PostPasswordResetRoute`, after SparkJava decodes, we still have an encoded value, causing client not found and the tests to fail.

My solution here is to not use `HttpUrl`, and use `RestAssured`'s own support for adding query params instead to avoid double-encoding.

Reply:
Wow--the double encoding thing is interesting.  Thanks for this thorough summary of the findings!